third_party/blink/renderer/core/brave_page_graph/graphml.cc

```diff
@@ -25,21 +25,50 @@ uint32_t graphml_index = 0;
 GraphMLAttr::GraphMLAttr(const GraphMLAttrForType for_value,
                          const String& name,
                          const GraphMLAttrType type)
-    : id_(++graphml_index), for_(for_value), name_(name), type_(type) {}
+    : id_(++graphml_index), for_(for_value), name_(name), type_(type) {
+  if (for_value == kGraphMLAttrForTypeAll) {
+    // Increment the global index one further, since this instance will
+    // claim the ids for both the node and the edge versions of this attr.
+    graphml_index += 1;
+  }
+}
 
 GraphMLId GraphMLAttr::GetGraphMLId() const {
   return "d" + base::NumberToString(id_);
```

Comment:
this can be optimized by storing the string as a class member instead of generating it each time.

Reply:
done and done